app/models/vm.rb

```diff
@@ -10,6 +10,23 @@ class Vm < VmOrTemplate
 
   include Operations
 
+  # Don't tack on type query for this class (instead we use scope)
+  def self.finder_needs_type_condition?
+    false
```

Comment:
I think if you change this to

```suggestion
    self < base_model
```

Then you don't need the inherited override at all